app/models/message.rb

```diff
@@ -135,6 +135,7 @@ class Message < ApplicationRecord
   after_create_commit :execute_after_create_commit_callbacks
 
   after_update_commit :dispatch_update_event
+  after_commit :reindex_for_search, if: :should_index?, on: [:create, :update]
```

Comment:
Shouldn't we do this on destroy as well?

Reply:
Yes, I’m thinking of handling it separately as a `bulk_delete_job` as part of the existing delete_object_job. This would help us avoid an avalanche of `ReindexV2` jobs in case of an account or inbox deletion with a large number of messages.